metals/src/main/scala/scala/meta/internal/pantsbuild/BloopPants.scala

```diff
@@ -261,9 +261,27 @@ private class BloopPants(
     )
     val isBaseDirectory =
       projects.iterator.filter(_.sources.nonEmpty).map(_.directory).toSet
+    // NOTE(olafur): generate synthetic projects to improve the file tree view
+    // in IntelliJ. Details: https://github.com/olafurpg/intellij-bsp-pants/issues/7
+    val syntheticProjects: List[C.Project] = sourceRoots.flatMap { root =>
+      if (isBaseDirectory(root.toNIO) || projects.exists(
+          p => p.sources.exists(_.getParent == root.toNIO)
```

Comment:
Does this line have an effect since we primarily use sources globs?

Reply:
Right, I fixed it, but I'm still not sure if it's enough